### PR TITLE
Simplify controllers and views, make everything more consistent; refactor RPC calls

### PR DESCRIPTION
**Pull request by iurimatias**

### CODE REVIEW

File: src/status/core.nim [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+  try:
+    let inputJSON = %* {
+      "jsonrpc": "2.0",
+      "method": methodName,
+      "params": %payload
+    }
+    echo inputJSON
+    result = $libstatus.callPrivateRPC($inputJSON)
+  except:
+    echo "error doing rpc request"
```
Comment (richard-ramos) [inline src/status/core.nim:22]: Use nim-chronicles   `error "Error doing RPC request", method=methodName payload=payload` ,

File: src/models/profile.nim [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  MailServer* = ref object
+    name*, endpoint*: string
+
+type Profile* = ref object
+    username*, identicon*: string
+
+proc toProfileModel*(obj: object): Profile =
+    result = Profile(username: obj.name, identicon: obj.photoPath)
+
+# proc newProfileModel*(): ProfileModel =
```
Comment (richard-ramos) [inline src/models/profile.nim:13]: :scissors:

File: src/models/node.nim [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+proc newNodeModel*(): NodeModel =
+  result = NodeModel()
+  result.events = createEventEmitter()
+
+proc delete*(self: NodeModel) =
+  discard
+
+proc sendRPCMessageRaw*(self: NodeModel, msg: string): string =
+  echo "sending RPC message"
```
Comment (richard-ramos) [inline src/models/node.nim:18]: Remove this echo, or use nim-chronicles

File: src/models/chat/chat_message.nim [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  result.userName = ""
+  result.message = ""
+  result.timestamp = "0"
+  result.identicon = ""
+  result.isCurrentUser = false
+
+proc toChatMessage*(payload: JsonNode): ChatMessage =
+  result = ChatMessage(
+    userName: payload["alias"].str,
+    # message: message.payload["lastMessage"]["text"].str,
```
Comment (richard-ramos) [inline src/models/chat/chat_message.nim:25]: Would this work?  `message: message["text"].str,`

**Comment by iurimatias** [comment]
this should have been simplify_common_models not simplify_common_models2, closing this and opening https://github.com/status-im/nim-status-client/pull/114

File: src/status/core.nim (hunk 1 above)
Comment (zah) [reply src/status/core.nim:22]: BTW, Chronicles has a short-cut when the property name matches a variable name. It's ok to say:

```nim
error "Error doing RPC request", method=methodName, payload
```

**Pull request closed without merging**